src/main/java/org/myrobotlab/service/interfaces/ServoControl.java

```diff
@@ -120,150 +168,249 @@
    */
-  double getPos();
+  Double getPos();
 
   /**
-   * the calculated mapper output for the servo - this is <b> ALWAYS ALWAYS in
-   * DEGREES !</b> because this method is used by the controller - and the
-   * controller needs a stable method &amp; a stable unit FIXME - not sure if
-   * this is a good thing to expose
+   * @return the current rest position value
+   */
+  Double getRest();
+
+  /**
+   * Return current speed if set - if speed/speed control is not being use it is
+   * null.
+   * 
+   * @return
+   */
+  Double getSpeed();
+
+  /**
+   * This value is for the ServoController to consume.
+   * 
+   * The calculated mapper output for the servo - this is <b> ALWAYS ALWAYS in
+   * DEGREES !</b> because this method is used by the servo controller - and the
+   * controller needs a stable method &amp; a stable unit
+   * 
+   * The is the output of the mapper, the servo controller may have its own
+   * global mapper, this one is unique for a specific Servo
+   * 
+   * This is position the mapper returns after the calculation of mapper
    * 
    * @return the target output position
    */
-  double getTargetOutput();
+  Double getTargetOutput();
 
-  double getMaxVelocity();
+  /**
+   * If currently attached to a controller
+   * 
+   * @return
+   */
+  Boolean isAttached();
 
-  double getVelocity();
+  /**
+   * is the servo currently sending pwm position control
+   * @return
+   */
+  Boolean isEnabled();
 
   /**
-   * set the pin of the servo this does not 'attach' energize the pin only set
-   * the pin value
+   * Returns true if mapper is inverted
    * 
-   * @param pin
-   *          the pin number for the servo
+   * @return
    */
-  void setPin(int pin);
+  Boolean isInverted();
 
   /**
-   * @return the current acceleration value
+   * This sets the servo's mapper explicitly
+   * 
+   * @param minX
+   *          - min input
+   * @param maxX
+   *          - max input
+   * @param minY
+   *          - min output
+   * @param maxY
+   *          - max output
    */
-  double getAcceleration();
+  void map(Double minX, Double maxX, Double minY, Double maxY);
+  
+  void map(Integer minX, Integer maxX, Integer minY, Integer maxY);
 
-  /*
-   * synchronizing servos together e.g. leftEye.sync(rightEye)
+  /**
+   * moveToBlocking is a basic move command of the servo - usually is 0 - 180
+   * valid range but can be adjusted and / or re-mapped with min / max and map
+   * commands
+   * 
+   * TODO - moveToBlocking - blocks until servo sends "ARRIVED_TO_POSITION"
+   * response
+   * 
+   * @param pos
+   *          - position to move to
+   * @return true (why?)
    */
-  void sync(ServoControl sc);
+  void moveToBlocking(Double pos);
+  
+  void moveToBlocking(Integer pos);
 
   /**
-   * unsync a servo
+   * control message publishing moveTo
    * 
    * @param sc
+   * @return
    */
-  void unsync(ServoControl sc);
+  ServoControl publishMoveTo(ServoControl sc);

```

Comment:
shouldn't this publish some sort of a servoEvent not a the actual servo itself...